.github/workflows/tests.yml

```diff
@@ -0,0 +1,60 @@
+name: Tests
+
+on: [push, pull_request]
+
+jobs:
+  py27:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 2.7
+      - name: Install Tox
+        run: pip install tox
+      - name: Run Tox
+        run: tox -e py27
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.8
+      - name: Install Tox
+        run: pip install tox
+      - name: Run Tox
+        run: tox -e cov
+      - name: Install pytest cov
+        run: pip install pytest-cov
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          fail_ci_if_error: true
```

Comment:
this may need adding codecov token for proper upload to codecov

Reply:
I think that public repos do not need a token. I tried to upload the codecov to my personal space in codecov.io and it worked okay. I have problems activating cdn-definitions in the release-engineering namespace though.

Reply:
I have looked into the codecov issue and if I understand correctly [this guide](https://docs.codecov.com/docs#step-2-get-the-repository-upload-token) , I don't need to explicitly activate the repo, it will be activated on the first upload of codecov data. 

I have tried it with a test repo and it seems that the first codecov upload activates the repo (the status on codecov.io changes to "No data available") and after the second upload we get the codecov results displayed. This all works without token.